pkg/lib/cockpit-components-table.jsx

```diff
@@ -75,9 +75,36 @@ export const ListingTable = ({
     let rows = tableRows;
 
     const [expanded, setExpanded] = useState({});
+    const [newItems, setNewItems] = useState([]);
+    const [currentRowsKeys, setCurrentRowsKeys] = useState([]);
     const [activeSortIndex, setActiveSortIndex] = useState(sortBy ? sortBy.index : 0);
     const [activeSortDirection, setActiveSortDirection] = useState(sortBy ? sortBy.direction : SortByDirection.asc);
 
+    useEffect(() => {
+        const getRowKeys = rows => {
+            const keys = [];
+
+            rows.forEach(row => {
+                if (row.props && row.props.key)
+                    keys.push(row.props.key);
+            });
+
+            return keys;
+        };
+
+        const new_keys = getRowKeys(rows);
```

Comment:
This needs to be checked if it does not misbehave on any place where we use `ListingTable`. And it only works if the component specifies `props.keys` - but they should as it is a good practice if nothing else.
But I think that we might condition this by some property like `animateNewItems` so that we don't enable it by default everywhere.